src/applications/MachineShopSimulator.java

```diff
@@ -47,32 +47,19 @@ boolean moveToNextMachine(Job theJob, SimulationResults simulationResults) {
      *
      * @return last job run on this machine
      */
-    Job changeState(int theMachine) {// Task on theMachine has finished,
-                                            // schedule next one.
-        Job lastJob;
-
-        if (machine[theMachine].isInactive()) {// in idle or change-over
-                                                    // state
-            lastJob = null;
-            // wait over, ready for new job
-            if (machine[theMachine].noJobQueued()) // no waiting job
-                eList.setFinishTime(theMachine, largeTime);
-            else {// take job off the queue and work on it
-                machine[theMachine].updateActiveJob();
-                machine[theMachine].setTotalWait(timeNow);
-                machine[theMachine].incNumTasks();
-                int t = machine[theMachine].nextTask();
-                eList.setFinishTime(theMachine, timeNow + t);
-            }
-        } else {// task has just finished on machine[theMachine]
-                // schedule change-over time
-            lastJob = machine[theMachine].getActiveJob();
-            machine[theMachine].setInactive();
-            eList.setFinishTime(theMachine, timeNow
-                    + machine[theMachine].getChangeTime());
+    Job changeState(int theMachine) {// Task on theMachine has finished, schedule next one.
+        Job activeJob = null;
+        int finishTime = largeTime;
+
+        if (machine[theMachine].isActive()) {// Machine is working on a job
+            activeJob = machine[theMachine].getActiveJob();
+            finishTime= timeNow + machine[theMachine].scheduleChangeOverTime();
```

Comment:
Nit: It's _minorly_ too bad that you repeat `timeNow + ` in both this line and line 58. No need to fix it here, but it's a thing to think about.

Reply:
We have thought about this and thought about possibly including this addition in the methods we extracted to machine, but we thought that it would be better done in MSS(we could possibly do something with it, but I don't think we will be able to gain much by doing that).